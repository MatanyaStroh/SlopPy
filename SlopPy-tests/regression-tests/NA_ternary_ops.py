# do a bunch of ternary operations on an NA object

x = 1 / 0
assert type(x) is NA

assert type(pow(x, 2)) is NA
assert type(pow(2, x)) is NA
assert type(x ** 2) is NA
assert type(2 ** x) is NA

