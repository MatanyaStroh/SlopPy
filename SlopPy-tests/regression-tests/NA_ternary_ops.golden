{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/NA_ternary_ops.py", line 3, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/NA_ternary_ops.py', '__doc__': None, '__package__': None}
}
{ternary_op(NA,*,*) | regression-tests/NA_ternary_ops.py <module>:6}
{ternary_op(*,NA,*) | regression-tests/NA_ternary_ops.py <module>:7}
{ternary_op(NA,*,*) | regression-tests/NA_ternary_ops.py <module>:8}
{ternary_op(*,NA,*) | regression-tests/NA_ternary_ops.py <module>:9}
